sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InMemoryWatermarkManager.java

```diff
@@ -427,22 +433,34 @@ public synchronized WatermarkUpdate refresh() {
       for (Watermark input : inputWms) {
         minTime = INSTANT_ORDERING.min(minTime, input.get());
       }
-      for (CommittedBundle<?> bundle : pendingBundles) {
-        // TODO: Track elements in the bundle by the processing time they were output instead of
-        // entire bundles. Requried to support arbitrarily splitting and merging bundles between
-        // steps
-        minTime = INSTANT_ORDERING.min(minTime, bundle.getSynchronizedProcessingOutputWatermark());
+      for (Instant hold : pendingValueHolds.values()) {
+        minTime = INSTANT_ORDERING.min(minTime, hold);
       }
       earliestHold.set(minTime);
       return WatermarkUpdate.fromTimestamps(oldHold, minTime);
     }
 
-    public synchronized void addPending(CommittedBundle<?> bundle) {
-      pendingBundles.add(bundle);
+    /**
+     * Add the elements contained within the {@link CommittedBundle bundle} to the collection of
+     * pending elements, holding the {@link SynchronizedProcessingTimeInputWatermark} to the
+     * synchronized processing time the bundle was committed.
```

Comment:
I thought we no longer had the synchronized processing time the bundle was comitted at?